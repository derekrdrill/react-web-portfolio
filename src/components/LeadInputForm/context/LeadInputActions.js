// TODO: implement axios here

// import axios from 'axios';
// const ADD_LEAD_INPUT = 'https://localhost:3001/addLeadInput';

// export const addLeadInput = axios.create({
//   baseURL: ADD_LEAD_INPUT,
//   headers: { 'Content-Type': 'application/json' },
// });

// export const postLeadInput = async leads => await addLeadInput.post(leads);

export const addLeadInput = async leads => {
  await fetch('http://localhost:3001/addLeadInput', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(leads),
  });
};
