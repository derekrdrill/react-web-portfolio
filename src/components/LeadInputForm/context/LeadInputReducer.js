export const leadInputReducer = (state, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
