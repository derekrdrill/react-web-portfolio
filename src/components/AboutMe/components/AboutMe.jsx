import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, List, ListItem } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ABOUT_ME_TEXT } from '../constants/ABOUT_ME_TEXT';
import bitmojiChillinWithBirds from '../../../assets/bitmoji_chillinWithBirds1.png';

export const AboutMe = ({ id }) => {
  // test change

  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <AboutMeContainer darkMode={darkMode} id={id} fluid>
        <Row>
          <Col className='py-3'>
            <AboutMeTitle className='text-center' component='h1' darkMode={darkMode} variant='h3'>
              ABOUT ME
            </AboutMeTitle>
          </Col>
        </Row>
        <Row>
          <AboutMeTextContainer
            darkMode={darkMode}
            xs={{ span: 10, order: 'last', offset: 1 }}
            lg={{ span: 5, order: 'first', offset: 1 }}
          >
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[1].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[2].text}
            </AboutMeText>
            <List style={{ color: 'white' }}>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[3].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[4].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[5].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[6].text}</AboutMeListItem>
            </List>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[7].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[8].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[9].text}
            </AboutMeText>
          </AboutMeTextContainer>
          <AboutMeBitmojiContainer xs={12} lg={6}>
            <img src={bitmojiChillinWithBirds} alt='Chillin with birds' />
          </AboutMeBitmojiContainer>
        </Row>
      </AboutMeContainer>
    </>
  );
};

const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  },
}));

export const AboutMeContainer = styled(Container)(({ darkMode }) => ({
  padding: '30px 50px 200px 50px',
  borderTop: '2px solid transparent',
  borderImage: darkMode
    ? 'linear-gradient(to right, rgba(248, 184, 255, 1), skyblue, gainsboro)'
    : 'linear-gradient(to right, rgba(176, 52, 197, 1), rgba(86, 206, 210, 1))',
  borderImageSlice: 1,
}));

export const AboutMeTitle = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'gainsboro' : 'darkslategray',
}));

export const AboutMeTextContainer = styled(Col)(({ darkMode }) => ({
  padding: 40,
  height: 550,
  overflowY: 'auto',
  border: '1px dashed #383838',
  borderRadius: 2,
  backgroundColor: darkMode ? '#303030' : '#eee5c1',
}));

export const AboutMeBitmojiContainer = styled(Col)({
  textAlign: 'center',
  paddingTop: 75,
  paddingBottom: 20,
});

export const AboutMeText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
  marginBottom: 15,
}));

export const AboutMeListItem = styled(ListItem)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
}));
