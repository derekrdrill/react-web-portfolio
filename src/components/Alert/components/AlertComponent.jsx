import React, { useContext } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Alert, AlertTitle } from '@mui/material';
import { AlertContext } from '../context/AlertContext';

export const AlertComponent = () => {
  const { alert, fadeOut } = useContext(AlertContext);

  return (
    alert && (
      <StyledAlert severity={alert.type} $fadeOut={fadeOut}>
        <AlertTitle>{alert.title}</AlertTitle>
        {alert.msg}
      </StyledAlert>
    )
  );
};

const fadeOut = keyframes({
  '0%': {
    opacity: 1,
  },
  '100%': {
    opacity: 0,
  },
});

const StyledAlert = styled(Alert)`
  animation-name: ${props =>
    props.$fadeOut
      ? css`
          ${fadeOut}
        `
      : 'none'};
  animation-duration: 500ms;
  animation-direction: linear;
`;
