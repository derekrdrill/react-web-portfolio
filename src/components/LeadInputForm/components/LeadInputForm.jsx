import React, { useState } from 'react';
import $ from 'jquery';
import styled from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, Button } from '@mui/material';

import { DynamicFormInputs } from '../../DynamicFormInputs/DynamicFormInputs';
import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';

import { addLeadInput } from '../context/LeadInputActions';

import { LEAD_INFO_INPUTS } from '../constants/LEAD_INFO_INPUTS';

export const LeadInputForm = () => {
  const [leads, setLeads] = useState({});
  const [loading, setLoading] = useState(null);

  const updateLeadsObject = e => {
    let key = e.target.id;
    let value = e.target.value;
    let updatedValue = { [key]: value };

    setLeads(leads => ({
      ...leads,
      ...updatedValue,
    }));
  };

  const postApi = async () => {
    setLoading(true);
    setTimeout(() => {
      addLeadInput(leads);
      setLoading(null);
      $('input').val('');
    }, 1000);
  };

  return (
    <Container>
      <Row>
        <Col xs={{ span: 6, offset: 3 }} lg={{ span: 4, offset: 3 }} className='mt-5'>
          <Typography variant='h6' component='h1'>
            Please fill in your information below
          </Typography>
        </Col>
      </Row>
      <InputsRow>
        <Col xs={{ span: 12 }} lg={{ span: 6, offset: 3 }}>
          <DynamicFormInputs inputs={LEAD_INFO_INPUTS} onChange={updateLeadsObject} />
        </Col>
      </InputsRow>
      <SubmitButtonRow>
        <Col xs={{ span: 12 }} lg={{ span: 1, offset: 8 }}>
          <Button onClick={postApi} variant='contained' fullWidth>
            Submit
          </Button>
        </Col>
      </SubmitButtonRow>
      <LoaderSpinner open={loading} />
    </Container>
  );
};


const InputsRow = styled(Row)({
  marginTop: '5%',
});

const SubmitButtonRow = styled(Row)({
  marginTop: '5%',
});
