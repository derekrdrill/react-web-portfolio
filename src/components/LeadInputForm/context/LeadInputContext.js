import React, { createContext, useReducer } from 'react';
import PropTypes from 'prop-types';

import { leadInputReducer } from './LeadInputReducer';

export const LeadInputContext = createContext();

const LeadInputProvider = ({ children }) => {
  const initialState = {
    page: 'form',
    tooltipOpen: false,
  };

  const [state, leadInputDispatch] = useReducer(leadInputReducer, initialState);

  return (
    <LeadInputContext.Provider value={{ leadInputDispatch, ...state }}>
      {children}
    </LeadInputContext.Provider>
  );
};

LeadInputProvider.propTypes = {
  children: PropTypes.node,
};

export default LeadInputProvider;


