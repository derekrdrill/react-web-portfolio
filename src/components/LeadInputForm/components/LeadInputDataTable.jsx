import React from 'react';
import { DynamicDataTable } from '../../DynamicDataTable/components/DynamicDataTable';

export const headers = [
  { headerID: 'firstName', headerName: 'First Name' },
  { headerID: 'lastName', headerName: 'Last Name' },
  { headerID: 'email', headerName: 'Email' },
  { headerID: 'phone', headerName: 'Phone Number' },
];

const getLeadsAPICall = 'http://localhost:3001/leadInput';
const deleteLeadsAPICall = 'http://localhost:3001/deleteLeadInput';
const editLeadsAPICall = 'http://localhost:3001/replaceLeadInput';

export const LeadInputDataTable = () => (
  <DynamicDataTable
    checkAllColor='primary'
    checkOneColor='secondary'
    deleteDataRowsAPICall={deleteLeadsAPICall}
    deleteDataRowNameItems={['firstName', ' ', 'lastName']}
    editDataRowsAPICall={editLeadsAPICall}
    getDataRowsAPICall={getLeadsAPICall}
    headers={headers}
    size='small'
  />
);
