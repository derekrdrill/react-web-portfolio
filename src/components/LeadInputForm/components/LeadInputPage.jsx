import React, { useContext } from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { AppBar, Toolbar, Button, Tooltip, Typography } from '@mui/material';
import { LeadInputContext } from '../context/LeadInputContext';
import { LeadInputForm } from './LeadInputForm';
import { LeadInputDataTable } from './LeadInputDataTable';

export const LeadInputPage = () => {
  const { leadInputDispatch, page, tooltipOpen } = useContext(LeadInputContext);

  const handlePageButtonClick = () => {
    const newPage = page === 'form' ? 'table' : 'form';
    leadInputDispatch({ type: 'SWITCH_PAGES', payload: { page: newPage, tooltipOpen: false } });
  };

  const handleTooltipToggle = () => leadInputDispatch({ type: 'TOGGLE_TOOLTIP', payload: !tooltipOpen });

  return (
    <>
      <PageBodyStyle />
      <LeadInputPageHeader>
        <Toolbar>
          <Container fluid>
            <Row>
              <ButtonContainer xs={12}>
                <Tooltip
                  open={tooltipOpen}
                  onClose={handleTooltipToggle}
                  onOpen={page === 'form' && handleTooltipToggle}
                  title={
                    <Typography variant='subtitle2' component='p'>
                      This would likely be an admin button in a real-life scenario
                    </Typography>
                  }
                >
                  <Button onClick={handlePageButtonClick}>
                    {page === 'form' ? 'See all records' : 'Go back to input form'}
                  </Button>
                </Tooltip>
              </ButtonContainer>
            </Row>
          </Container>
        </Toolbar>
      </LeadInputPageHeader>
      <LeadInputPageContent>
        {page === 'form' && <LeadInputForm />}
        {page === 'table' && <LeadInputDataTable />}
      </LeadInputPageContent>
    </>
  );
};

const PageBodyStyle = createGlobalStyle({
  body: {
    backgroundColor: 'tan',
  },
});

const LeadInputPageHeader = styled(AppBar)({
  top: 80,
  boxShadow: 'none',
  backgroundColor: 'inherit',
});

const LeadInputPageContent = styled.div({
  marginTop: '10%',
});

const ButtonContainer = styled(Col)({
  textAlign: 'right',
});
