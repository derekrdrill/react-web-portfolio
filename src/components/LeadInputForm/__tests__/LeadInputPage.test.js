import 'regenerator-runtime/runtime';

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { LeadInputPage } from '../components/LeadInputPage';
import { LeadInputProvider } from '../context/LeadInputContext';

const renderer = new ShallowRenderer();

describe('Lead Input Page tests', () => {
  it('renders correctly', () => {
    renderer.render(
      <LeadInputProvider>
        <LeadInputPage />
      </LeadInputProvider>,
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
