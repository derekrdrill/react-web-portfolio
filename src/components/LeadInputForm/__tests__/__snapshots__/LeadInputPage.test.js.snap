// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lead Input Page tests renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "leadInputDispatch": [Function],
      "page": "form",
      "tooltipOpen": false,
    }
  }
>
  <LeadInputPage />
</Context.Provider>
`;
